src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,63 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final String COMMA = ",";
+    private static final int OPERATION_INDEX = 0;
+    private static final int VALUE_INDEX = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        Map<String, Integer> operationTotals = calculateTotals(fromFileName);
+        int supply = operationTotals.getOrDefault(SUPPLY, 0);
+        int buy = operationTotals.getOrDefault(BUY, 0);
+        int result = supply - buy;
+        writeToFile(toFileName, supply, buy, result);
+        printToConsole(supply, buy, result);
```

Comment:
use this template to implement SOLID principles
```suggestion
        String dataFromFile = readFile(fromFileName);
        String report = createReport(dataFromFile);
        writeReportToFile(report, toFileName);
```